src/internal/operators/take.ts

```diff
@@ -45,26 +46,50 @@ import { OperatorSubscriber } from './OperatorSubscriber';
  * the source if the source emits fewer than `count` values.
  */
 export function take<T>(count: number): MonoTypeOperatorFunction<T> {
-  return count <= 0
-    ? // If we are taking no values, that's empty.
-      () => EMPTY
-    : operate((source, subscriber) => {
-        let seen = 0;
-        source.subscribe(
-          new OperatorSubscriber(subscriber, (value) => {
-            // Increment the number of values we have seen,
-            // then check it against the allowed count to see
-            // if we are still letting values through.
-            if (++seen <= count) {
-              subscriber.next(value);
-              // If we have met or passed our allowed count,
-              // we need to complete. We have to do <= here,
-              // because re-entrant code will increment `seen` twice.
-              if (count <= seen) {
-                subscriber.complete();
-              }
-            }
-          })
-        );
-      });
+  if (count < 1) {
+    // If we are taking no values, that's empty.
+    return () => EMPTY;
+  }
+
+  if (count === Number.POSITIVE_INFINITY) {
```

Comment:
```suggestion
  if (count === Infinity) {
```

Reply:
is there a preference for the global Infinity? its the same value as `Number.POSITIVE_INFINITY`, personally I prefer the latter